dts/vendor/particle/mesh_feather.dtsi

```diff
@@ -0,0 +1,147 @@
+/*
```

Comment:
Maybe I missed this discussion but why are those in `dts/vendor` instead of `boards/arm/<board>`?

Reply:
Excellent; I was wondering if anybody'd be concerned.  This is in the PR description:
> This is the first set of boards that share device tree information. I've chosen to put that material into dts/vendor/particle, as it was possible to include from there without messing with cmake. If this should be changed let me know.

I don't love the solution, but I like the alternatives less (duplication in each board directory, or hackery to add multiple include paths for just these boards).  I'd hoped for wider attention to this PR allowing the innovation to be discussed.

Reply:
Apologies for not having noticed this in the PR description. I am happy with this as a solution personally, maybe not thrilled about `vendor` as a folder name. We already have "vendors" in each architecture under `dts/<arch>` so maybe this should be in `dts/boards`?

Reply:
Paging @galak, @agross-linaro, @ioannisg and @MaureenHelm to get their input on this matter.

Reply:
I tagged this at TSC so we are reminded of this outstanding decision during the meeting.

Reply:
I feel like I'd rather see this done in the board dir and not in the dts/ dir.

I think we should see if we can improve how BOARD_ARCH_DIR is set in cmake/app/boilerplate.cmake so that we could do something like:

boards/arm/particle/argon
boards/arm/particle/xenon
boards/arm/particle/common

etc.

Reply:
@galak that'd be fine, but I can't do it (I don't speak CMake), so this PR would have to be held until somebody else gets around to it and the new capability is merged so this can be reworked to use it.

Reply:
Ok, I've implemented the start of this in PR #12377